app/src/main/java/com/github/sdp_begreen/begreen/fragments/ProfileDetailsFragment.kt

```diff
@@ -0,0 +1,102 @@
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.RatingBar
+import android.widget.TextView
+import androidx.lifecycle.lifecycleScope
+import com.github.sdp_begreen.begreen.R
+import com.github.sdp_begreen.begreen.User
+import kotlinx.coroutines.launch
+
+
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+const val ARG_USER = "USER"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [ProfileDetailsFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class ProfileDetailsFragment : Fragment() {
+    var user: User? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            user = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                it.getParcelable(ARG_USER, User::class.java)
+            }else {
+                it.getParcelable(ARG_USER)
+            }
+        }
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+
+        val view = inflater.inflate(R.layout.fragment_profile_details, container, false)
+        val name: TextView = view?.findViewById(R.id.fragment_profile_details_profile_name) as TextView
+        val rating: RatingBar = view.findViewById(R.id.fragment_profile_details_profile_rating) as RatingBar
+        val profileImgView: ImageView = view.findViewById(R.id.fragment_profile_details_profile_image) as ImageView
+        val profileDescription: TextView = view.findViewById(R.id.fragment_profile_details_profile_description) as TextView
+        val profilePhone: TextView = view.findViewById(R.id.fragment_profile_details_profile_phone) as TextView
+        val profileEmail: TextView = view.findViewById(R.id.fragment_profile_details_profile_email) as TextView
+        val userTextLevel: TextView = view.findViewById(R.id.fragment_profile_details_level) as TextView
+        val userProgressBar: ProgressBar = view.findViewById(R.id.fragment_profile_details_user_progress) as ProgressBar
+        val followButton: Button = view.findViewById(R.id.fragment_profile_details_follow_button) as Button
+        name.text = user?.name
+        rating.rating = user?.score?.toFloat() ?: 0.0f
+        profileImgView.setImageBitmap(user?.img?.getPhotoFromDataBase())
+        profileDescription.text = user?.description
+        profilePhone.text = user?.phone
+        profileEmail.text = user?.email
+        userTextLevel.text = user?.name + " level's"
+        userProgressBar.progress = user?.progression ?: 0
+
+        followButton.setOnClickListener {
+            if(followButton.text == "Follow") {
```

Comment:
Here I think that we could create an `enum` for the value `Follow` and `Unfollow` to better express what that represents.

Something like a `Action` `enum` that will encapsulate actions

Reply:
Oh yeah I like that proposition!